src/mozilla_sec_eia/models/sec10k/ex_21/inference.py

```diff
@@ -79,9 +79,8 @@ def create_inference_dataset(pdfs_dir: Path, labeled_json_dir=None, has_labels=F
     else:
         inference_df = format_unlabeled_pdf_dataframe(pdfs_dir=pdfs_dir)
     image_dict = get_image_dict(pdfs_dir)
-    doc_filenames = inference_df["id"].unique()
     annotations = []
-    for filename in doc_filenames:
+    for filename in image_dict:
```

Comment:
Filenames are keys in `image_dict`, so it's a bit more straightforward to just loop through them.